src/ccsr/src/client.rs

```diff
@@ -85,6 +86,46 @@ impl Client {
         })
     }
 
+    /// Gets the latest schema for the specified subject as well as all other subjects referenced
+    /// by that subject (recursively).
+    ///
+    /// The dependencies are returned in alphabetical order by subject name.
+    pub async fn get_all_subjects(
+        &self,
+        subject: &str,
+    ) -> Result<(Subject, Vec<Subject>), GetBySubjectError> {
+        let mut subjects = VecDeque::new();
+        let mut seen = HashSet::new();
+        let mut subjects_queue = vec![(subject.to_owned(), "latest".to_owned())];
+        while let Some((subject, version)) = subjects_queue.pop() {
+            let req = self.make_request(Method::GET, &["subjects", &subject, "versions", &version]);
+            let res: GetBySubjectResponse = send_request(req).await?;
+            subjects.push_back(Subject {
+                schema: Schema {
+                    id: res.id,
+                    raw: res.schema,
+                },
+                version: res.version,
+                name: res.subject.clone(),
+            });
+            seen.insert(res.subject);
+            if let Some(r) = res.references {
+                subjects_queue.extend(
+                    r.into_iter()
+                        .filter(|r| !seen.contains(&r.subject))
+                        .map(|r| (r.subject, r.version.to_string())),
+                );
+            }
+        }
+
+        let primary = subjects
+            .pop_front()
+            .expect("Request should error if no subjects");
+        let mut dependencies: Vec<_> = subjects.into();
```

Comment:
I'm not sure the `VecDeque` actually buys you anything here; if you pop from the front, I _think_ the conversion to a `Vec` is gonna have to do some shuffling. So might be a bit simpler to just make `subjects` a `Vec` to start and just `remove(0)` to get `primary`.

Reply:
Ahh yeah sure: https://doc.rust-lang.org/src/alloc/collections/vec_deque/mod.rs.html#2955